parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -497,7 +523,30 @@ pub mod pallet {
 			let beacon_block_root: H256 =
 				merkleization::hash_tree_root_beacon_header(header.clone())
 					.map_err(|_| Error::<T>::HeaderHashTreeRootFailed)?
-					.into(); // TODO check denylist headers
+					.into();
+
+			let finalized_block_root_hash =
+				<FinalizedBeaconHeadersBlockRoot<T>>::get(last_finalized_header.beacon_block_root);
+
+			let max_slots_per_historical_root = T::MaxSlotsPerHistoricalRoot::get();
+			let depth = max_slots_per_historical_root.ilog2();
+			let index_in_array = block_slot % max_slots_per_historical_root as u64;
+			let leaves_start_index = 2u32.pow(depth as u32) as u64;
+			let leaf_index: u64 = (leaves_start_index + index_in_array) as u64;
```

Comment:
same as https://github.com/Snowfork/snowbridge/pull/765/files#r1105271929

and also not sure we can use exp/log functions in `no_std` runtime

Reply:
Same answer as here: https://github.com/Snowfork/snowbridge/pull/765#discussion_r1105337368

I'll check the no-std usage, just wanted to get it working for now. 😄

Reply:
I've added these as parachain config: https://github.com/Snowfork/snowbridge/pull/765/files#diff-20b75e7903673f591cfb5dd3c153d51797bd4069c9b71dfb6b11e06f2c8c675eR7